src/storage.rs

```diff
@@ -13,6 +14,29 @@ use super::errors::Error;
 pub type DatabaseConnection = r2d2::PooledConnection<SqliteConnectionManager>;
 pub type DatabaseConnectionPool = r2d2::Pool<SqliteConnectionManager>;
 
+#[derive(PartialEq, Eq, Hash)]
+pub struct RoomId {
+    id: String,
+}
+
+impl RoomId {
+    pub fn new(room_id: &str) -> RoomId {
+        let whitespace = Regex::new(r"/\s+/g").expect("regex is valid");
```

Comment:
You probably want to create new regex objects every time this functions is called. You could follow this advise to make it static: https://docs.rs/regex/1.5.4/regex/#example-avoid-compiling-the-same-regex-in-a-loop